### PR TITLE
Fix for #121 and #157

### LINKED ISSUE #121: Make proper link in notification
Hi,

when clicking on "Open app" in a kresus notification ("x operations imported"), it would be nice to directly go to the given account if possible.

### LINKED ISSUE #157: Display the account balance in the left menu
Hello @bnjbvr,

When I connect to Kresus, I have to click on each account to see their balances. It would be better to see all of them in the left menu where account name are displayed. What do you think about it?

### PR DESCRIPTION
**Pull request by ZeHiro**
Make proper file link fixes #121
Display account amount in side bar fixes #157

### CODE REVIEW

**Comment by bnjbvr** [comment]
Thanks a lot for the PR! I'll add a few review comments, but I can merge your PR and fix them myself, if you don't have time to do it. I am wondering how come this branch can have conflicts? Can you make sure you're working on the latest incoming branch, please, for the next time?
(also, if you don't mind next time: split your commits, so that each commit maps to a single feature and fixes a single issue. Thanks!)

File: client/components/AccountList.js [hunk 1]
```diff
@@ -13,13 +13,17 @@ class AccountListItem extends React.Component {
     onClick() {
         Actions.SelectAccount(this.props.account);
     }
-
+    ComputeTotal(operations) {
+        var total = operations
+                        .reduce((a,b) => a + b.amount, this.props.account.initialAmount);
+        return Math.round (total * 100) / 100;
```
Comment (bnjbvr) [inline client/components/AccountList.js]: space before the opening paren

File: client/components/AccountList.js [hunk 2]
```diff
@@ -13,13 +13,17 @@ class AccountListItem extends React.Component {
...
+    }
     render() {
         var maybeActive = this.props.active ? "active" : "";
         return (
             <li className={maybeActive}>
                 <span>
-                    <a href="#" onClick={this.onClick.bind(this)}>{this.props.account.title}</a>
+                    <a href="#" onClick={this.onClick.bind(this)}>{this.props.account.title}</a> {this.ComputeTotal(this.props.account.operations)} €
```
Comment (bnjbvr) [inline client/components/AccountList.js]: I'd put the amount between parenthesis

File: client/components/AmountWell.js [hunk 3]
```diff
@@ -19,7 +19,7 @@ export class AmountWell extends React.Component {
         var total = operations
                         .filter(this.props.filterFunction)
                         .reduce((a,b) => a + b.amount, this.props.initialAmount);
-        return (total * 100 | 0) / 100;
+        return Math.round (total * 100) / 100;
```
Comment (bnjbvr) [inline client/components/AmountWell.js]: space before the opening paren

File: client/components/BankList.js [hunk 4]
```diff
@@ -16,7 +16,9 @@ class BankListItemComponent extends React.Component {
     onClick() {
         Actions.SelectBank(this.props.bank);
     }
-
+    
+    
+
```
Comment (bnjbvr) [inline client/components/BankList.js]: three blank lines

File: client/components/OperationList.js [hunk 5]
```diff
@@ -71,8 +71,12 @@ class CategorySelectComponent extends React.Component {
     }
 }
 
-function ComputeAttachmentLink(opId) {
-    return `operations/${opId}/file`;
+function ComputeAttachmentLink(op) {
+    let file = 'file';
+    if (op.binary.fileName !== null) {
```
Comment (bnjbvr) [inline client/components/OperationList.js]: if (typeof op.binary.fileName !== 'null')

File: server/controllers/operations.coffee [hunk 6]
```diff
@@ -67,17 +67,23 @@ module.exports.file = (req, res, next) ->
...
-            res.sendStatus stream.statusCode
+    BankOperation.find req.params.bankOperationID, (err, operation) =>
+        if err?
+            h.sendErr res, 'when retrieving bank operation'
+            return
+        request = http.get options, (stream) ->
+
+            if stream.statusCode is 200
+                # XXX note that this is dependent upon the real type, which could
+                # be different from pdf...
```
Comment (bnjbvr) [inline server/controllers/operations.coffee]: stall comment, can be removed

File: server/controllers/operations.coffee [hunk 7]
```diff
@@ -67,17 +67,23 @@ module.exports.file = (req, res, next) ->
...
+                fileMime = 'application/pdf'
+                if operation.binary.fileMime?
+                    fileMime = operation.binary.fileMime
```
Comment (bnjbvr) [inline server/controllers/operations.coffee]: fileMime = operation.binary.fileMime
fileMime ?= 'application/pdf';

File: client/components/OperationList.js (hunk 5 above)
Comment (ZeHiro) [inline client/components/OperationList.js]: I don't see why :disappointed: , is it a cleaner way to detect there is no "fileName" parameter form binary?

File: client/components/AccountList.js (hunk 2 above)
Comment (ZeHiro) [inline client/components/AccountList.js]: OK

**Comment by ZeHiro** [comment]
I'll make the corrections tonight. And I don't remember if I was completly up to date with the inconming branch

File: client/components/OperationList.js (hunk 5 above)
Comment (bnjbvr) [inline client/components/OperationList.js]: Oops, I was confused with `undefined`, which should also probably be tested here. A quicker test (but error-prone, if the fileName variable is falsy) would be `let file = op.binary.fileName || 'file';`, which is fine to me too.

**Comment by ZeHiro** [comment]
Good to me. Maybe it also fixes #141 as I changed the way of rouding the amount displayed (use of Math.round) I'll tell on my setup when merged

**Comment by bnjbvr** [comment]
Fantastic, thank you! I'll try to do the manual conflict resolution this week, so as to get it merged soon.

**Comment by ZeHiro** [comment]
Potential conflicts are in routes.coffee, OperationList.js.

**Comment by bnjbvr** [comment]
It's now merged in `incoming`, which means it is going to be in the next released version (probably out in September). Thanks a lot for this contribution!

**Pull request merged**